jcl/src/java.base/share/classes/java/lang/Access.java

```diff
@@ -600,6 +603,21 @@ public char getUTF16Char(byte[] val, int index) {
 	public int countPositives(byte[] ba, int off, int len) {
 		return StringCoding.countPositives(ba, off, len);
 	}
+
+	@PreviewFeature(feature = PreviewFeature.Feature.STRING_TEMPLATES)
```

Comment:
What's the point of the annotation on this and the next two methods?

Reply:
This is prepare for [JEP 430: String Templates (Preview)](https://openjdk.org/jeps/430)

Reply:
They have meaning for the interface definition; but I don't see any value here. Please explain why the annotations are necessary.

Reply:
This is to match openjdk implementation.
I think `javac` processes it - https://github.com/ibmruntimes/openj9-openjdk-jdk/blob/31c58a953edb55b0a3a4fe5fbae4ff17f8eb8810/src/jdk.compiler/share/classes/com/sun/tools/javac/comp/TypeEnter.java#L343C3-L344

Reply:
I expect that only applies to code that _uses_ string templates, and isn't relevant to our implementation of JavaLangAccess. We'll only have to remove the annotations later (once the feature is out of preview); I don't see any value to adding them here in the first place.

Reply:
The three `JavaLangAccess` APIs annotated with `STRING_TEMPLATES` are openjdk `StringBuilder` usages, I expect openjdk uses string templates in those cases assuming the annotation was added for such purpose.

Reply:
The only uses I see (so far) are in `java.util.FormatItem`; only those uses are affected by the annotations on the interface methods; they have no visibility of the implementations.

Reply:
Removed `@PreviewFeature` annotation from `Access` implementation.

Also added `JVM_PrintWarningAtDynamicAgentLoad()` temporary implementation which was just introduced by latest `jdk-21+25`.